### PR TITLE
Updated dependencies for Python 3.12 compatibility.

### PR DESCRIPTION
**Pull request by qwrrty**
I'm fiddling with the code to see if I can add Bluesky support. As part of that work, I needed to update it to a newer version of Python (Pillow 3 would not work under anything newer than Python 3.5).

Here are the changes I made to run this under Python 3.12, in case you're interested:

* Removed explicit version dependencies in requirements.txt to permit pip to resolve them itself.
* Dropped the `styles` argument from `bleach.clean()` which no longer takes this parameter as of v5.0.0
* Changed Selenium WebDriver from PhantomJS to Chrome (PhantomJS support was removed from Selenium 4.0.0)
* Some changes to permit a TESTING run when Twitter credentials are not set

Selenium recommends using either Chrome or Firefox in headless mode now that PhantomJS is no longer supported. I don't really care strongly which, I'm happy to follow your lead.

Note that I do not have a twitter API key and so have not tested that the twitter integration all still works with these changes.

### CODE REVIEW

**Comment by j-e-d** [comment]
Thanks for the PR. I will take a look into it later in the week. I started working on Bluesky and Mastodon support the last time they blocked most bots at twitter but never got to finish so I didn't update the repo. Someone donated an invite so the user is ready once the code is working, 
If you would rather use an account you own for this just let me know so we don't duplicate the content over there.

**Comment by qwrrty** [comment]
Thanks for the quick reply! I did set up a Bluesky account to experiment with this, and I'll continue tinkering to see if I can help get it done, but I have no strong feelings about ownership :-)

File: requirements.txt
Comment (j-e-d) [inline requirements.txt:1]: Would you mind changing this to use the current version that pip resolved to as minimum version and set the next semver major number as maximum?

e.g.

alembic>=1.13.1<2.0.0
bleach>=6.1.0<7.0.0

etc.

**Review by j-e-d — CHANGES_REQUESTED** [review]
The code looks good, just a change on the pip versioning to avoid any surprises on major changes on the used packages.

File: requirements.txt
Comment (qwrrty) [reply requirements.txt:1]: Oh, I like that approach, thanks for that suggestion! Added.

**Review by j-e-d — APPROVED** [review]
thanks

**Pull request merged**